pkg/pipelines/bootstrap.go

```diff
@@ -197,3 +220,40 @@ func marshalOutputs(out io.Writer, outputs []interface{}) error {
 	}
 	return nil
 }
+
+// validateImageRepo validates the input image repo.  It determines if it is
+// for internal registry and prepend internal registry hostname if neccessary.
+func validateImageRepo(o *BootstrapOptions) (bool, string, error) {
+	components := strings.Split(o.ImageRepo, "/")
+	errorMsg := "failed to parse image repo:%s, expected image repository in the form <registry>/<username>/<repository> or <project>/<app> for internal registry"
```

Comment:
I'd probably just extract this to a function that generated the error with a message?

Reply:
This function validates and retrieves image repo from the options.   We can change the function name to "getImageRepo"   I am not quite sure I am following what you are saying.   Please clarify.

Reply:
Sorry, the errorMsg could just be 

```go
func validationError(imageRepo string) error {
  return fmt.Errorf("failed to parse image repo: %s, expected image repository in the form <registry>/<username>/<repository> or <project>/<app> for internal registry", imageRepo)
}
```

Reply:
Fixed.